src/v2-router.js

```diff
@@ -0,0 +1,228 @@
+  }
+
+  if (body.addressCounty) {
+    cleanedBody.addressCounty = body.addressCounty.trim();
+  }
+
+  if (body.addressPostcode) {
+    cleanedBody.addressPostcode = body.addressPostcode.trim();
+  }
+
+  if (body.phoneNumber) {
+    cleanedBody.phoneNumber = body.phoneNumber.trim();
+  }
+
+  if (body.emailAddress) {
+    cleanedBody.emailAddress = utils.formatters.stripAndRemoveObscureWhitespace(body.emailAddress.toLowerCase());
+  }
+
+  return cleanedBody;
+};
+
+// #region Health Check
+
+/**
+ * Gets the health status message for the API.
+ */
+v2Router.get('/health', async (request, response) => {
+  response.status(200).send({message: 'OK'});
+});
+
+// #endregion
+
+// #region Registrations - Listing
+
+/**
+ * READs all the registrations in the application.
+ */
+v2Router.get('/registrations', async (request, response) => {
+  return response.status(501).send({message: 'Not implemented.'});
+});
+
+// #endregion
+
+// #region Registrations - Single Entity
+
+/**
+ * CREATEs a single registration.
+ */
+v2Router.post('/registrations', async (request, response) => {
+  return response.status(501).send({message: 'Not implemented.'});
+});
+
+/**
+ * READs a single registration.
+ */
+v2Router.get('/registrations/:id', async (request, response) => {
+  return response.status(501).send({message: 'Not implemented.'});
+});
+
+/**
+ * UPDATEs a single registration.
+ */
+v2Router.put('/registrations/:id', async (request, response) => {
+  return response.status(501).send({message: 'Not implemented.'});
+});
+
+/**
+ * UPDATEs part of a single registration.
+ */
+v2Router.patch('/registrations/:id', async (request, response) => {
+  try {
+    // Try to parse the incoming ID to make sure it's really a number.
+    const existingId = Number(request.params.id);
+    if (Number.isNaN(existingId)) {
+      return response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+    }
+
+    // Check if there's a registration allocated at the specified ID.
+    const existingReg = await Registration.findOne(existingId);
+    if (existingReg === undefined || existingReg === null) {
+      return response.status(404).send({message: `Registration ${existingId} not allocated.`});
+    }
+
+    // Clean up the user's input before we store it in the database.
+    const cleanObject = cleanPatchInput(request.body);
+
+    // Update the registration in the database with our client's values.
+    const updatedReg = await Registration.update(existingId, cleanObject);
+
+    // If they're not successful, send a 500 error.
+    if (updatedReg === undefined) {
+      return response.status(500).send({message: `Could not update registration ${existingId}.`});
+    }
+
+    // If they are, send back the updated fields.
+    return response.status(200).send(updatedReg);
+  } catch (error) {
+    // If anything goes wrong (such as a validation error), tell the client.
+    return response.status(500).send({error});
+  }
```

Comment:
You're already catching errors here, so just throwing a couple of validation errors should be fine!

Reply:
Resolved.